internal/log_analysis/log_processor/parsers/awslogs/cloudtrail_digest.go

```diff
@@ -0,0 +1,104 @@
+	DigestPublicKeyFingerprint  *string                   `json:"digestPublicKeyFingerprint" validate:"required" description:"The hexadecimal encoded fingerprint of the public key that matches the private key used to sign this digest file."`
+	DigestSignatureAlgorithm    *string                   `json:"digestSignatureAlgorithm" validate:"required" description:"The algorithm used to sign the digest file."`
+	LogFiles                    []CloudTrailDigestLogFile `json:"logFiles" validate:"required,min=0" description:"Log files delivered in this digest"`
+
+	// NOTE: added to end of struct to allow expansion later
+	AWSPantherLog
+}
+
+// nolint:lll
+type CloudTrailDigestLogFile struct {
+	S3Bucket        *string            `json:"s3Bucket" validate:"required" description:"The name of the Amazon S3 bucket for the log file."`
+	S3Object        *string            `json:"s3Object" validate:"required" description:"The Amazon S3 object key of the current log file."`
+	HashValue       *string            `json:"hashValue" validate:"required" description:"The hexadecimal encoded hash value of the uncompressed log file content."`
+	HashAlgorithm   *string            `json:"hashAlgorithm" validate:"required" description:"The hash algorithm used to hash the log file."`
+	NewestEventTime *timestamp.RFC3339 `json:"newestEventTime" validate:"required" description:"The UTC time of the most recent event among the events in the log file."`
+	OldestEventTime *timestamp.RFC3339 `json:"oldestEventTime" validate:"required" description:"The UTC time of the oldest event among the events in the log file."`
+}
+
+type CloudTrailDigestParser struct{}
+
+// NOTE: guard to ensure interface implementation
+var _ parsers.LogParser = (*CloudTrailDigestParser)(nil)
+
+func (p *CloudTrailDigestParser) New() parsers.LogParser {
+	return &CloudTrailDigestParser{}
+}
+
+// Parse returns the parsed events or nil if parsing failed
+func (p *CloudTrailDigestParser) Parse(log string) ([]*parsers.PantherLog, error) {
+	event := &CloudTrailDigest{}
+	err := jsoniter.UnmarshalFromString(log, event)
+	if err != nil {
+		return nil, err
+	}
+
+	event.updatePantherFields(p)
+
+	if err := parsers.Validator.Struct(event); err != nil {
+		return nil, err
+	}
+	return event.Logs(), nil
+}
+
+// LogType returns the log type supported by this parser
+func (p *CloudTrailDigestParser) LogType() string {
+	return "AWS.CloudTrailDigest"
+}
+
+func (event *CloudTrailDigest) updatePantherFields(p *CloudTrailDigestParser) {
+	// Use end time as it's the time the digest was actually delivered
+	event.SetCoreFields(p.LogType(), event.DigestEndTime, event)
+	event.AppendAnyAWSAccountIdPtrs(event.AWSAccountID)
+	event.AppendAnySHA256HashesPtr(event.PreviousDigestHashValue)
+	for i := range event.LogFiles {
+		lf := &event.LogFiles[i]
+		event.AppendAnySHA256HashesPtr(lf.HashValue)
+	}
```

Comment:
You could make that slightly more compact: 
```suggestion
	for _, file := range event.LogFiles {
		event.AppendAnySHA256HashesPtr(file.HashValue)
	}
```

Reply:
Will change, it's more readable that way. It's an optimization habit, getting just the pointer is faster in slices of `[]Struct` because `for _, foo := range foos` copies the struct to the stack so if the slice element is *not* a pointer it's a tiny bit heavier